lib/Frontend/CompilerInvocation.cpp

```diff
@@ -680,10 +681,10 @@ bool FrontendArgsToOptionsConverter::computeFallbackModuleName() {
 }
 
 bool FrontendArgsToOptionsConverter::computeOutputFilenames() {
-  assert(
-      FrontendOptions::doesActionProduceOutput(Opts.RequestedAction) ||
-      !FrontendOptions::doesActionProduceTextualOutput(Opts.RequestedAction));
-
+  if (!FrontendOptions::doesActionProduceOutput(Opts.RequestedAction)) {
+    Opts.OutputFilenames.clear();
```

Comment:
Is it possible to assert this instead? Or does someone actually provide an output filename and expect it to be cleared?

Reply:
Interesting thought. I'll try asserting it at the start of computeOutputFilenames.